src/margo-api-reference/workload-api/application-package-api/application-description.linkml.yaml

```diff
@@ -583,3 +725,45 @@ slots:
     inlined_as_list: true
     rank: 20
 
+enums:
+  CpuArchitectureType:
+    permissible_values:
+      amd64:
+        description: AMD 64-bit architecture.
+      x86_64:
+        description: x86 64-bit architecture.
+      arm64:
+        description: ARM 64-bit architecture.
+      arm:
+        description: ARM 32-bit architecture.
+  
+  CommunicationInterfaceType:
```

Comment:
I'd add something like "other" to be able to cope with future upcoming interfaces at a specification level. I'd then add an attribute/slot of type "string" called "other_type" (or so) to enable future extensions.